jobs/base_model_validator_test.py

```diff
@@ -0,0 +1,172 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Unit tests for BaseModelValidator"""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import datetime
+import unittest
+
+import apache_beam as beam
+from apache_beam.runners.direct import direct_runner
+from apache_beam.testing import test_pipeline as pipeline
+from apache_beam.testing import util as beam_testing_util
+
+
+from core.platform import models
+from jobs import base_model_validator
+from jobs import base_model_validator_errors as errors
+
+(base_models, user_models) = models.Registry.import_models(
+    [models.NAMES.base_model, models.NAMES.user])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class MockModel(base_models.BaseModel):
+    pass
+
+
+class BaseModelValidatorTests(unittest.TestCase):
+
+    def setUp(self):
+        self.now = datetime.datetime.utcnow()
+        self.year_ago = self.now - datetime.timedelta(weeks=52)
+        self.year_later = self.now + datetime.timedelta(weeks=52)
+
+    def test_base_model_validator_ptransform(self):
+        with pipeline.TestPipeline(runner=direct_runner.DirectRunner()) as p:
+            invalid_id = MockModel(
+                id='123@?!*',
+                deleted=False,
+                created_on=self.year_ago,
+                last_updated=self.now)
+            invalid_timestamp = MockModel(
+                id='124',
+                deleted=False,
+                created_on=self.now,
+                last_updated=self.year_later)
+            expired_model = MockModel(
+                id='123',
+                deleted=True,
+                created_on=self.year_ago,
+                last_updated=self.year_ago)
+            valid_model = MockModel(
+                id='123',
```

Comment:
Please use a distinct ID for each model.

Reply:
done